bookwyrm/views/inbox.py

```diff
@@ -130,15 +130,13 @@ def has_valid_signature(request, activity):
     """verify incoming signature"""
     try:
         signature = Signature.parse(request)
-
-        key_actor = urldefrag(signature.key_id).url
-        if key_actor != activity.get("actor"):
-            raise ValueError("Wrong actor created signature.")
-
-        remote_user = activitypub.resolve_remote_id(key_actor, model=models.User)
+        remote_user = activitypub.resolve_remote_id(activity.get("actor"), model=models.User)
         if not remote_user:
             return False
 
+        if signature.key_id != remote_user.key_pair.remote_id:
```

Comment:
The only downside of this is that it ...completely breaks compatibility between Bookwyrm instances because of the signatures problem. 🙃 Whoops. Might need a workaround.

Reply:
Where we set the user's key does have a slash, but currently when we create the signature we state it doesn't. So the proposed fix is to add the slash when signing.

I had a similar thought but it will absolutely break backwards compatibility because user keys are created when the user is created.

On the other hand, I strongly suspect that user key ids ending in `/#key-id` wasn't originally intended because it's kind of weird. If we have to account for multiple possibilities anyway it might be ok to change new keys rather than new signatures and then make the signature a slightly different way 🤔

Reply:
Hmm ok now I have woken up properly, what I wrote makes less sense. 

If we fix the signing mechanism to match the existing key creation code it will break backwards-compatabiity of signature-creation. This eventually will resolve itself by people ugrading, and in the meantime with some additional checks.

If we fix the key-creation code so that new users _don't_ have the slash in the key id, this will break backwards-compatibility of user key IDs, which can only be resolved by running a DB query to change every existing user's keys.